osu.Framework/Bindables/BindableDictionary.cs

```diff
@@ -0,0 +1,589 @@
+                DisabledChanged?.Invoke(disabled);
+        }
+
+        #endregion ICanBeDisabled
+
+        #region IUnbindable
+
+        public void UnbindEvents()
+        {
+            CollectionChanged = null;
+            DisabledChanged = null;
+        }
+
+        public void UnbindBindings()
+        {
+            if (bindings == null)
+                return;
+
+            foreach (var b in bindings)
+            {
+                Debug.Assert(b != null);
+                b.unbind(this);
+            }
+
+            bindings?.Clear();
+        }
+
+        public void UnbindAll()
+        {
+            UnbindEvents();
+            UnbindBindings();
+        }
+
+        public void UnbindFrom(IUnbindable them)
+        {
+            if (!(them is BindableDictionary<TKey, TValue> tThem))
+                throw new InvalidCastException($"Can't unbind a bindable of type {them.GetType()} from a bindable of type {GetType()}.");
+
+            removeWeakReference(tThem.weakReference);
+            tThem.removeWeakReference(weakReference);
+        }
+
+        private void unbind(BindableDictionary<TKey, TValue> binding)
+        {
+            Debug.Assert(bindings != null);
+            bindings.Remove(binding.weakReference);
+        }
+
+        #endregion IUnbindable
+
+        #region IHasDescription
+
+        public string? Description { get; set; }
+
+        #endregion IHasDescription
+
+        #region IBindableCollection
+
+        void IBindable.BindTo(IBindable them)
+        {
+            if (!(them is BindableDictionary<TKey, TValue> tThem))
+                throw new InvalidCastException($"Can't bind to a bindable of type {them.GetType()} from a bindable of type {GetType()}.");
+
+            BindTo(tThem);
+        }
+
+        void IBindableDictionary<TKey, TValue>.BindTo(IBindableDictionary<TKey, TValue> them)
+        {
+            if (!(them is BindableDictionary<TKey, TValue> tThem))
+                throw new InvalidCastException($"Can't bind to a bindable of type {them.GetType()} from a bindable of type {GetType()}.");
+
+            BindTo(tThem);
+        }
+
+        /// <summary>
+        /// An alias of <see cref="BindTo"/> provided for use in object initializer scenarios.
+        /// Passes the provided value as the foreign (more permanent) bindable.
+        /// </summary>
+        public BindableDictionary<TKey, TValue> BindTarget
+        {
+            set => ((IBindableDictionary<TKey, TValue>)this).BindTo(value);
+        }
+
+        /// <summary>
+        /// Binds this <see cref="BindableDictionary{TKey, TValue}"/> to another.
+        /// </summary>
+        /// <param name="them">The <see cref="BindableDictionary{TKey, TValue}"/> to be bound to.</param>
+        public void BindTo(BindableDictionary<TKey, TValue> them)
+        {
+            if (them == null)
+                throw new ArgumentNullException(nameof(them));
+            if (bindings?.Contains(weakReference) ?? false)
```

Comment:
My personal preference wrt readability is to write:
```suggestion
            if (bindings?.Contains(weakReference) == true)
```